src/ast/passes/codegen_llvm.cpp

```diff
@@ -867,14 +871,29 @@ void CodegenLLVM::visit(Call &call)
     b_.CreateMemsetBPF(buf,
                        b_.getInt8(0),
                        bpftrace_.config_.get(ConfigKeyInt::max_strlen));
-    call.vargs.front()->accept(*this);
+    const uint64_t max_size = bpftrace_.config_.get(ConfigKeyInt::max_strlen);
+    Value *sz;
+    if (call.vargs.size() > 1) {
+      auto &arg = *call.vargs.at(1);
+      auto scoped_del = accept(&arg);
+      Value *pr_sz = b_.CreateIntCast(expr_, b_.getInt32Ty(), false);
```

Comment:
Probably doesn't make much of a difference but it seems like "str" uses a Int64: `expr_ = b_.CreateIntCast(expr_, b_.getInt64Ty(), true);`

Reply:
Ah it's cause the path function is accepting an int32. It's fine, leave as you have it.